hubblestack/extmods/fdg/command_line_parser.py

```diff
@@ -0,0 +1,193 @@
+                continue
+
+            regex_list = []
+            regex_pattern = "(?<=(\s|{))-{0,2}\"{0,1}\'{0,1}" + key_alias + "\"{0,1}\'{0,1}\s*" + delimiter + "\s*openingbracket.*closingbracket(?=(\s|$))"
+            braces_list = [('\[','\]'), ('\{','\}'), ('\(','\)')]
+            for item in braces_list:
+                regex = re.sub("openingbracket", item[0], regex_pattern)
+                regex = re.sub("closingbracket", item[1], regex)
+                regex_list.append(regex)
+            regex1 = "(?<=(\s|{))-{0,2}\"{0,1}\'{0,1}" + key_alias + "\"{0,1}\'{0,1}\s*" + delimiter + "\s*([\",\']).*?\\2"
+            regex2 = "(?<=(\s|{))-{0,2}\"{0,1}\'{0,1}" + key_alias + "\"{0,1}\'{0,1}\s*" + delimiter + "\s*.+?(?=(\s|$))"
+            regex_list.append(regex1)
+            regex_list.append(regex2)
+
+            for regex in regex_list:
+                log.debug("looping with regex : %s", regex)
+                match_list = _get_match_list(regex, key_alias, command_line, delimiter)
+                if match_list:
+                    ret_match_list.extend(match_list)
+                    break
+
+        return True, ret_match_list
+    except Exception as e:
+        log.exception("Some exception occurred in command_line_parser's parse_cmdline function %s", e)
+        return False, ''
+
+
+def _get_match_list(regex, key_alias, command_line, delimiter):
+    """
+
+    :param regex: search for this pattern in command_line
+    :param key_alias: The key whose value is to be found out
+    :param command_line: command_line from where the value is to be fetched
+    :param delimiter: assignment operator between key and value
+    :return: List of all matching patterns
+
+    This function will fetch all matching patterns of regex in command_line and will then strip the
+    matched value to remove the leading and trailing hyphens (-), <spaces> and quotes.
+    Since the regex also contains the 'key' and 'delimiter', that is also stripped away to get the final value.
+    """
+    match_list = []
+
+    matches = re.finditer(regex, command_line)
+    for matchNum, match in enumerate(matches, start=1):
+        log.info("Match {matchNum} was found at {start}-{end}: {match}".format(matchNum=matchNum,
+                                                                                   start=match.start(), end=match.end(),
+                                                                                   match=match.group()))
+        value = match.group().lstrip("-")
+        value = value.replace("\"" + key_alias + "\"", key_alias)
+        value = value.replace("\'" + key_alias + "\'", key_alias)
+        prefix = key_alias + "\s*" + delimiter + "\s*"
+        value = re.sub(prefix, '', value)
+        value = value.rstrip('\"\' ')
+        value = value.lstrip('\"\' ')
+        if value[0] in open_bracket_list:
+            value = _fetch_bracketed_value(value)
```

Comment:
Please add error logging here in case empty string is returned from _fetch_bracketed_value function

Reply:
Done.